### PR TITLE
Fix endless loop when opening streams with missing information

### PR DESCRIPTION
**Pull request by peak3d**
see title

## Description
CDemuxClient evaluates a few parameters in ParsePacket.
On change the streams are again requested and the corrected values are overwritten.
This leads to an endless loop inside stream selection.

This issue was introduced with https://github.com/xbmc/xbmc/pull/12469

## How Has This Been Tested?
netflix addon / win32

## Types of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp [hunk 1]
```diff
@@ -369,6 +369,7 @@ void CDVDDemuxClient::RequestStream(CDemuxStream *stream, std::map<int, std::sha
   }
 
   std::shared_ptr<CDemuxStream> dStream = GetStreamInternal(stream->uniqueId);
+  bool updateExitsingStream(dStream != nullptr);
```
Comment (hudokkow) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp]: hum... `updateExitingStream`?

Comment (peak3d) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp]: ????

Comment (peak3d) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp]: ah! thx

Comment (hudokkow) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp]: ;)

**Comment by mase76** [comment]
I confirm this bug as solved with this patch.

**Comment by Rechi** [comment]
jenkins build this please

**Comment by MilhouseVH** [comment]
Confirmed here too, tested Amazon/Netflix on RPi3 and i5 NUC. All good now.

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by peak3d** [comment]
not ready to merge - need some more comments / renamings

**Comment by MilhouseVH** [comment]
I can confirm your "plane swap" commit fixes the red/blue colour swap when using the "old" `libwidevinecdm.so`.

File: xbmc/cores/VideoPlayer/Process/VideoBuffer.h [hunk 2]
```diff
@@ -99,7 +99,7 @@ class CVideoBuffer
   virtual uint8_t* GetMemPtr() { return nullptr; };
   virtual void GetPlanes(uint8_t*(&planes)[YuvImage::MAX_PLANES]) {};
   virtual void GetStrides(int(&strides)[YuvImage::MAX_PLANES]) {};
-  virtual void SetDimensions(int width, int height, const int (&strides)[YuvImage::MAX_PLANES]) {};
+  virtual void SetDimensions(int width, int height, const int (&strides)[YuvImage::MAX_PLANES], const int planeOffsets[YuvImage::MAX_PLANES]) {};
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.h]: please keep the style and pass a ref, not a pointer

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.h]: why don't you keep the method with 3 arguments and add another with 4

Comment (peak3d) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.h]: Its not as ref because is optional. You can leave it nullptr and have the previous plane offset calculation. Sure I could make a new call with 4, but this would lead to not small code duplication in CVideoBufferSysMem.

Comment (peak3d) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.h]: code has moved to an own PR: https://github.com/xbmc/xbmc/pull/12494

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/Process/VideoBuffer.h]: > but this would lead to not small code duplication in CVideoBufferSysMem

does not require any kind of code dupe. just call the 3-arguments method and set data pointer with offsets.
